### PR TITLE
change default malloc method from ptmalloc to tcmalloc

### PR DESCRIPTION
**Pull request by qwdarrenyin**
change default malloc method from ptmalloc to tcmalloc

### CODE REVIEW

File: cmake/ProjectTcMalloc.cmake [hunk 1]
```diff
@@ -0,0 +1,37 @@
+include(ExternalProject)
+
+set(TCMALLOC_CONFIG ./autogen.sh COMMAND ./configure CXXFLAGS=-DHAVE_POSIX_MEMALIGN_SYMBOL=1 --enable-frame-pointers --disable-cpu-profiler --disable-heap-profiler --disable-heap-checker --disable-debugalloc --enable-minimal)
```
Comment (bxq2011hust) [inline cmake/ProjectTcMalloc.cmake]: Please refer here
https://github.com/gperftools/gperftools/blob/fe62a0baab87ba3abca12f4a621532bf67c9a7d2/README#L39

and here 

https://github.com/gperftools/gperftools/blob/fe62a0baab87ba3abca12f4a621532bf67c9a7d2/README#L46-L47

File: cmake/ProjectTcMalloc.cmake [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ExternalProject_Add(gptcmalloc
+    PREFIX ${CMAKE_SOURCE_DIR}/deps/
+    DOWNLOAD_NAME at8_0-release.zip
+    DOWNLOAD_NO_PROGRESS 1
+    URL https://github.com/gperftools/gperftools/archive/at8_0-release.zip
+    URL_HASH SHA256=16041e6b52fe082805d7847ba9fab21f7ce97ecb1c6d1fd99c3130137f8bcf91
+    BUILD_IN_SOURCE 1
+    LOG_CONFIGURE 0
+    LOG_BUILD 0
+    LOG_INSTALL 0
```
Comment (bxq2011hust) [inline cmake/ProjectTcMalloc.cmake]:
```suggestion
    LOG_CONFIGURE 1
    LOG_BUILD 1
    LOG_INSTALL 1
```

File: cmake/ProjectTcMalloc.cmake (hunk 1 above)
Comment (bxq2011hust) [reply cmake/ProjectTcMalloc.cmake]: --disable-shared

Comment (bxq2011hust) [reply cmake/ProjectTcMalloc.cmake]: why do we need `autogen.sh`? It seems official tutorial doesn't include this step?
https://github.com/gperftools/gperftools/blob/fe62a0baab87ba3abca12f4a621532bf67c9a7d2/INSTALL#L376-L393

File: cmake/ProjectTcMalloc.cmake [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    set(TCMALLOC_MAKE make)
+else()
+    set(TCMALLOC_MAKE make)
+endif()
+
+ExternalProject_Add(gptcmalloc
+    PREFIX ${CMAKE_SOURCE_DIR}/deps/
+    DOWNLOAD_NAME at8_0-release.zip
+    DOWNLOAD_NO_PROGRESS 1
+    URL https://github.com/gperftools/gperftools/archive/at8_0-release.zip
```
Comment (bxq2011hust) [inline cmake/ProjectTcMalloc.cmake]: why use 2.21? the newest is 2.7 29 Apr 2018.

File: cmake/ProjectTcMalloc.cmake (hunk 2 above)
Comment (qwdarrenyin) [reply cmake/ProjectTcMalloc.cmake]: already change

File: cmake/ProjectTcMalloc.cmake (hunk 1 above)
Comment (qwdarrenyin) [reply cmake/ProjectTcMalloc.cmake]: already add --disable-shared and --disable-test

File: cmake/ProjectTcMalloc.cmake [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    set(TCMALLOC_MAKE make)
+else()
+    set(TCMALLOC_MAKE make)
+endif()
+
+ExternalProject_Add(gptcmalloc
+    PREFIX ${CMAKE_SOURCE_DIR}/deps/
+    DOWNLOAD_NAME master.zip
+    DOWNLOAD_NO_PROGRESS 1
+    URL https://github.com/gperftools/gperftools/archive/master.zip
```
Comment (bxq2011hust) [inline cmake/ProjectTcMalloc.cmake]:
```suggestion
    URL https://github.com/gperftools/gperftools/releases/download/gperftools-2.7/gperftools-2.7.tar.gz
```

File: cmake/ProjectTcMalloc.cmake [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+if (APPLE)
+    set(TCMALLOC_MAKE make)
+else()
+    set(TCMALLOC_MAKE make)
+endif()
+
+ExternalProject_Add(gptcmalloc
+    PREFIX ${CMAKE_SOURCE_DIR}/deps/
+    DOWNLOAD_NAME master.zip
```
Comment (bxq2011hust) [inline cmake/ProjectTcMalloc.cmake]:
```suggestion
    DOWNLOAD_NAME gperftools-2.7.tar.gz
```

File: .travis.yml [hunk 6]
```diff
@@ -68,13 +72,14 @@ addons:
...
     - pip
+    - automake
 install: |
     if [[ $TRAVIS_OS_NAME == linux && "$TRAVIS_BRANCH" != "master" ]]; then 
       bash tools/ci/check-commit.sh; 
     fi
 script: |
     mkdir -p build && cd build && CC=/usr/bin/clang CXX=/usr/bin/clang++ cmake .. && \
-    make -j2 && cd ../tools && bash ci/ci_check.sh
+    make && cd ../tools && bash ci/ci_check.sh
```
Comment (bxq2011hust) [inline .travis.yml]:
```suggestion
    make -j2 && cd ../tools && bash ci/ci_check.sh
```
I think, this should not be changed.

File: cmake/ProjectTcMalloc.cmake [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    LOG_BUILD 1
+    LOG_INSTALL 1
+    CONFIGURE_COMMAND ${TCMALLOC_CONFIG} 
+    BUILD_COMMAND ${TCMALLOC_MAKE}
+    INSTALL_COMMAND ""
+    BUILD_BYPRODUCTS <SOURCE_DIR>/.libs/libtcmalloc_minimal.a
+)
+
+ExternalProject_Get_Property(gptcmalloc SOURCE_DIR)
+add_library(tcmalloc STATIC IMPORTED GLOBAL)
```
Comment (bxq2011hust) [inline cmake/ProjectTcMalloc.cmake]:
```suggestion
add_library(TCMalloc STATIC IMPORTED GLOBAL)
```
what about using it's official name, as same as others

**Pull request merged**